prow/plugins/trigger/trigger.go

```diff
@@ -119,42 +123,51 @@ func getClient(pc plugins.PluginClient) client {
 }
 
 func handlePullRequest(pc plugins.PluginClient, pr github.PullRequestEvent) error {
-	trustedOrg, joinOrgURL := trustedOrgForRepo(pc.PluginConfig, pr.Repo.Owner.Login, pr.Repo.Name)
-	return handlePR(getClient(pc), trustedOrg, joinOrgURL, pr)
+	org, repo, _ := orgRepoAuthor(pr.PullRequest)
+	return handlePR(getClient(pc), pc.PluginConfig.TriggerFor(org, repo), pr)
 }
 
 func handleIssueComment(pc plugins.PluginClient, ic github.IssueCommentEvent) error {
-	trustedOrg, _ := trustedOrgForRepo(pc.PluginConfig, ic.Repo.Owner.Login, ic.Repo.Name)
-	return handleIC(getClient(pc), trustedOrg, ic)
+	return handleIC(getClient(pc), pc.PluginConfig.TriggerFor(ic.Repo.Owner.Login, ic.Repo.Name), ic)
 }
 
 func handlePush(pc plugins.PluginClient, pe github.PushEvent) error {
 	return handlePE(getClient(pc), pe)
 }
 
-// trustedOrgForRepo returns the configured trusted organization and a URL for it
-// for the provided org and repo combination.
-func trustedOrgForRepo(config *plugins.Configuration, org, repo string) (string, string) {
-	if trigger := config.TriggerFor(org, repo); trigger != nil && trigger.TrustedOrg != "" {
-		return trigger.TrustedOrg, trigger.JoinOrgURL
+// trustedUser returns true if user is trusted in repo.
+//
+// Trusted users are either repo collaborators, org members or trusted org members.
+// Whether repo collaborators and/or a second org is trusted is configured by trigger.
+func trustedUser(ghc githubClient, trigger *plugins.Trigger, user, org, repo string) (bool, error) {
+	// First check if user is a collaborator, assuming this is allowed
+	allowCollaborators := trigger == nil || !trigger.OnlyOrgMembers
+	if allowCollaborators {
+		if ok, err := ghc.IsCollaborator(org, repo, user); err != nil {
+			return false, fmt.Errorf("error in IsCollaborator: %v", err)
+		} else if ok {
+			return true, nil
+		}
 	}
-	return org, fmt.Sprintf("https://github.com/orgs/%s/people", org)
-}
 
-func isUserTrusted(ghc githubClient, user, trustedOrg, org string) (bool, error) {
-	orgMember, err := ghc.IsMember(trustedOrg, user)
-	if err != nil {
-		return false, err
-	} else if orgMember {
+	// Next see if the user is an org member
+	if member, err := ghc.IsMember(org, user); err != nil {
```

Comment:
If `allowCollaborators` is true we will check if the user is a collaborator of the repo and if they are not, then check if the user is a member of the org.
If the user is not a repo collaborator doesn't that imply that they won't be an org member?

Reply:
https://developer.github.com/v3/orgs/#edit-an-organization

The default permission for org members can be `None` in which case an org member may not have access to a private repo.

Most likely we should not schedule testing in this case, which is yet another reason why we should eventually retire this check (and the `TrustedOrg` field) in favor of using repo collaborators exclusively.